README.md

```diff
@@ -29,3 +33,50 @@ spark-client.spark-submit --deploy-mode cluster --conf spark.kubernetes.containe
 ```bash
 spark-client.spark-submit  --deploy-mode cluster --name pyspark-s3a --properties-file <path to spark-defaults.conf> --conf spark.kubernetes.container.image='localhost:32000/sparkrock:latest' <S3 location of pyspark script>
 ```
+
+## Use Spark from within a Container!
+
+The Canonical spark-client scripts are bundled within the OCI rock image as well. 
+To be able to launch Spark jobs in the Kubernetes cluster from within a container, only following additional resources are needed.
+
+- Kubernetes Configuration File at ```$HOME/.kube/config```.
+- Spark and S3 etc. Configuration.
+
+Follow these steps to 
+
+1. launch the container. 
+
+```bash
+export SPARK_DRIVER_PORT=20002
+export SPARK_BLOCKMANAGER_PORT=6060
+export SPARK_UID=185
+export SPARK_GID=185
+export SPARK_USER_HOME=/home/spark
+export SPARK_DRIVER_HOST=$(hostname -I | cut -d' ' -f1)
+docker run -it -p $SPARK_DRIVER_PORT:$SPARK_DRIVER_PORT -p $SPARK_BLOCKMANAGER_PORT:$SPARK_BLOCKMANAGER_PORT -u $SPARK_GID:$SPARK_UID -w $SPARK_USER_HOME --entrypoint /bin/bash docker.io/dataplatformoci/spark:3.3.1
+```
+
+2. From within the container, setup and launch the Spark job
+```bash
+# create kubeconfig
+mkdir .kube
+cat > .kube/config << EOF
+KUBECONFIG_CONTENTS
```

Comment:
I know the proper content here, but maybe we should explain users how to fill this?
The same for SPARK_CONF_CONTENTS below.

BTW. `mkdir -p $HOME/.kube` can be part of rockcraft.yaml. AFAIK it doesn't create empty folders, but you cat create empty `$HOME/.kube/config` as well.

P.S. this can stay AS IS.

Reply:
Thanks Alex, will keep in mind for future.